draft-irtf-cfrg-hybrid-kems.md

```diff
@@ -361,31 +436,31 @@ populated by concrete instantiations:
 * `XOF`: the eXtended Output Function
 * `PQKEM`: the PQ KEM component scheme
 * `G`: the nomimal group used to construct the traditional KEM component
-  scheme as described in {{group}
+  scheme as described in {{group}}
 * Nseed: length in bytes of the seed randomness sourced from the RNG
-* Npqseed: length in bytes of the input to PQ.DeriveKey()
+* Npqseed: length in bytes of the input to PQ.DeriveKeyPair()
 * Ntradseed: length in bytes of the input to NominalGroup.ScalarFromBytes()
 
 ~~~
-def expandDecapsulationKey(sk):
-  expanded = XOF(sk, Npqseed + Ntradseed)
-  (pq_PK, pq_SK) = PQKEM.DeriveKey(expanded[..Npqseed])
-  trad_SK = G.ScalarFromBytes(expanded[Npqseed..])
-  trad_PK = G.SerializeElement(NominalGroup.ScalarMultBase(trad_SK))
-  return (pq_SK, trad_SK, pq_PK, trad_PK)
-
-def KeyGen():
-  sk = random(Nseed)
-  (pq_SK, trad_SK, pq_PK, trad_PK) = expandDecapsulationKey(sk)
-  return sk, concat(pq_PK, trad_PK)
+def expandDecapsulationKey(dk):
+  expanded = XOF(dk, Npqseed + Ntradseed)
+  (pq_EK, pq_DK) = PQKEM.DeriveKeyPair(expanded[..Npqseed])
+  trad_DK = G.ScalarFromBytes(expanded[Npqseed..])
+  trad_EK = G.SerializeElement(NominalGroup.ScalarMultBase(trad_DK))
```

Comment:
This is breaking abstraction.

Reply:
Yep, this will get fixed in the next PR.  The only reason this part is getting touched is the global `(sk, pk) -> (dk, ek)` terminology change.